src/session_impl.cpp

```diff
@@ -6500,6 +6526,24 @@ namespace aux {
 		m_alerts.emplace_alert<dht_pkt_alert>(pkt, len, d, node);
 	}
 
+	bool session_impl::on_dht_request(char const* query, int query_len
+		, dht::msg const& request, entry& response)
+	{
+#ifndef TORRENT_DISABLE_EXTENSIONS
+		if (query_len > max_dht_query_length) return false;
+
+		for (m_extension_dht_queries_t::iterator i = m_extension_dht_queries.begin();
```

Comment:
It scales linearly with the number of registered extension queries. Currently it is called for all requests, I can of course change that if you want.

I designed it the way I did to maximize cache efficiency. I expect the number of registered queries will always be small enough that memory access time swamps the asymptotic gains of using a map.